onadata/libs/utils/logger_tools.py

```diff
@@ -501,6 +502,12 @@ def publish_form(callback):
             'text': _((u'An error occurred while publishing the form. '
                        'Please try again.')),
         }
+    except BadStatusLine:
+        # Catch urllib error
+        return {
+            'type': 'alert-error',
+            'text': _(u'An error occured while trying to retrieve the form.')
+        }
```

Comment:
What would the action that would be required by the user to take if they encounter this error?
Would the error message in https://github.com/onaio/onadata/pull/1765/files#diff-018f1a90f318bf1748393be886fee2c0R502-R503 be sufficient in this case? If it would be sufficient could we add `BadStatusLine` to `except (BadStatusLine, MemoryError, OSError)`?

Reply:
The error message would be perfect for this scenario. I'll update my PR.